aspnetcore/signalr/api-design.md

```diff
@@ -0,0 +1,78 @@
+
+For example, consider a server-side API like the following:
+
+[!code-csharp[ParameterBasedOldVersion](api-design/Samples.cs?name=ParameterBasedOldVersion)]
+
+The JavaScript client calls this method using `invoke` as follows:
+
+[!code-typescript[CallWithOneParameter](api-design/Samples.ts?name=CallWithOneParameter)]
+
+If you later add a second parameter to the server method, older clients won't provide this parameter value. For example:
+
+[!code-csharp[ParameterBasedNewVersion](api-design/Samples.cs?name=ParameterBasedNewVersion)]
+
+When the old client tries to invoke this method, it will get an error like this:
+
+```
+Microsoft.AspNetCore.SignalR.HubException: Failed to invoke 'GetTotalLength' due to an error on the server.
+```
+
+On the server, you'll see a log message like this:
+
+```
+System.IO.InvalidDataException: Invocation provides 1 argument(s) but target expects 2.
+```
+
+The old client only sent one parameter, but the newer server API required two parameters. Using custom objects as parameters gives you more flexibility. Let's redesign the original API to use a custom object:
+
+[!code-csharp[ObjectBasedOldVersion](api-design/Samples.cs?name=ObjectBasedOldVersion)]
+
+Now, the client uses an object to call the method:
+
+[!code-typescript[CallWithObject](api-design/Samples.ts?name=CallWithObject)]
+
+Instead of adding a parameter, add a property to the `GetTotalLengthRequest` object:
+
+[!code-csharp[ObjectBasedNewVersion](api-design/Samples.cs?name=ObjectBasedNewVersion&highlight=31,36-41)]
+
+When the old client sends a single parameter, the extra `Param2` property will be left `null`. You can detect a message sent by an older client by checking the `Param2` for `null` and apply a default value. A new client can send both parameters.
+
+[!code-typescript[CallWithObjectNew](api-design/Samples.ts?name=CallWithObjectNew)]
+
+The same technique works for methods defined on the client. You can send a custom object from the server side:
+
+[!code-csharp[ClientSideObjectBasedOld](api-design/Samples.cs?name=ClientSideObjectBasedOld)]
+
+On the client side, you access the `Message` property rather than using a parameter:
+
+[!code-typescript[OnWithObjectOld](api-design/Samples.ts?name=OnWithObjectOld)]
+
+If you later decide to add the sender of the message to the payload, add a property to the object:
+
+[!code-csharp[ClientSideObjectBasedNew](api-design/Samples.cs?name=ClientSideObjectBasedNew&highlight=60)]
+
+The older clients won't be expecting the `Sender` value, so they'll ignore it. A new client can accept it by updating to read the new property:
+
+[!code-typescript[OnWithObjectNew](api-design/Samples.ts?name=OnWithObjectNew&highlight=21-24)]
```

Comment:
I don't see the highlight for any of your snippets

Reply:
Me neither ☹️

Reply:
Maybe `highlight` isn't supported with `name`? I think I specified everything right: https://dotnet.github.io/docfx/spec/docfx_flavored_markdown.html?tabs=tabid-1%2Ctabid-a#code-snippet

Reply:
@scottaddie do you know what might be going on here?

Reply:
Not seeing anything obviously wrong here, but I'll continue to dig. On a related note, can you please move the sample code files into a signalr/api-design/sample folder? That's the convention we've followed elsewhere.

Reply:
I see the problem now. Numbering resets within each region. In the case of the OnWithObjectNew region, your highlighting numbers should be 2-5.